include/secp256k1_frost.h

```diff
@@ -0,0 +1,695 @@
+ *  In:    nonce: pointer to the nonce
+ */
+SECP256K1_API int secp256k1_frost_pubnonce_serialize(
+    const secp256k1_context* ctx,
+    unsigned char *out66,
+    const secp256k1_frost_pubnonce* nonce
+) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3);
+
+/** Serialize a FROST partial signature
+ *
+ *  Returns: 1 when the signature could be serialized, 0 otherwise
+ *  Args:    ctx: a secp256k1 context object
+ *  Out:   out32: pointer to a 32-byte array to store the serialized signature
+ *  In:      sig: pointer to the signature
+ */
+SECP256K1_API int secp256k1_frost_partial_sig_serialize(
+    const secp256k1_context* ctx,
+    unsigned char *out32,
+    const secp256k1_frost_partial_sig* sig
+) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3);
+
+/** Parse a FROST partial signature.
+ *
+ *  Returns: 1 when the signature could be parsed, 0 otherwise.
+ *  Args:    ctx: a secp256k1 context object
+ *  Out:     sig: pointer to a signature object
+ *  In:     in32: pointer to the 32-byte signature to be parsed
+ *
+ *  After the call, sig will always be initialized. If parsing failed or the
+ *  encoded numbers are out of range, signature verification with it is
+ *  guaranteed to fail for every message and public key.
+ */
+SECP256K1_API int secp256k1_frost_partial_sig_parse(
+    const secp256k1_context* ctx,
+    secp256k1_frost_partial_sig* sig,
+    const unsigned char *in32
+) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3);
+
+/** Serialize a FROST share
+ *
+ *  Returns: 1 when the share could be serialized, 0 otherwise
+ *  Args:    ctx: a secp256k1 context object
+ *  Out:   out32: pointer to a 32-byte array to store the serialized share
+ *  In:    share: pointer to the share
+ */
+SECP256K1_API int secp256k1_frost_share_serialize(
+    const secp256k1_context* ctx,
+    unsigned char *out32,
+    const secp256k1_frost_share* share
+) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3);
+
+/** Parse a FROST share.
+ *
+ *  Returns: 1 when the share could be parsed, 0 otherwise.
+ *  Args:    ctx: a secp256k1 context object
+ *  Out:   share: pointer to a share object
+ *  In:     in32: pointer to the 32-byte share to be parsed
+ */
+SECP256K1_API int secp256k1_frost_share_parse(
+    const secp256k1_context* ctx,
+    secp256k1_frost_share* share,
+    const unsigned char *in32
+) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3);
+
+/** Creates a key generation share and verifiable secret sharing ("VSS")
+ *  commitments.
+ *
+ *  To generate a key, each participant generates a share for each other
+ *  participant. For example, in the case of 2 particpants, Alice and Bob, they
+ *  each generate 2 shares, distribute 1 share to each other using a secure
+ *  channel, and keep 1 for themselves.
+ *
+ *  Each participant _must_ have a secure channel with each other participant
```

Comment:
I think the expected properties of what is a "secure channel" could be defined (e.g authentication, encryption, replay, downgrade), or this could be precised in `frost.md`. If libsecp256k1 has already a definition of a secure channel somewhere else, a reference can be made. If we're just expecting authentication/encryption, example of framework fulfilling those purposes can be given (e.g Noise with the right pattern).

Reply:
Great point, I will add this.

Reply:
@elichai That's an interesting idea. It would be great to remove the requirement of a secure channel. Do you have any thoughts about what encryption we might use? Perhaps something built on `secp256k1_scalar_chacha20`?

Reply:
I initially replied outside this thread and then deleted it and added my comment here, so linking up @elichai's response here: https://github.com/ElementsProject/secp256k1-zkp/pull/138#issuecomment-1382094686